src/generate/register.rs

```diff
@@ -771,9 +771,9 @@ impl Variant {
         let span = Span::call_site();
         evs.values
             .iter()
-            // filter out all reserved variants, as we should not
+            // filter out all reserved or default variants, as we should not
             // generate code for them
-            .filter(|field| field.name.to_lowercase() != "reserved")
+            .filter(|field| field.name.to_lowercase() != "reserved" && field.is_default == None)
```

Comment:
This will fail in some cases as noted. A suggestion is to run svd2rust-regress and then diff the outputs, that can help with determining validity of change.